benchmarks/ab_benchmark.py

```diff
@@ -0,0 +1,272 @@
+@click.option('--content_type', '-ic', default='application/jpg', help='content type')
+@click.option('--image', '-di', default='', help='custom docker image')
+@click.option('--docker_runtime', '-dr', default='', help='docker runtime')
+@click.option('--ts', '-ts', type=click.BOOL, default=False, help='use already running docker instace')
+@click_config_file.configuration_option(provider=json_provider, help="read configuration from a JSON file")
+def ab_benchmark(preset, url, device, exec_env, concurrency, requests, batch_size, batch_delay, input_file, workers,
+                 content_type, image, docker_runtime, ts):
+    input_params = {'url': url,
+                    'device': device,
+                    'exec_env': exec_env,
+                    'batch_size': batch_size,
+                    'batch_delay': batch_delay,
+                    'workers': workers,
+                    'concurrency': concurrency,
+                    'requests': requests,
+                    'input': input_file,
+                    'content_type': content_type,
+                    'image': image,
+                    'docker_runtime': docker_runtime
+                    }
+
+    # set ab params
+    preset_ab_run[preset]()
+    update_exec_params(input_params)
+    click.secho("Starting AB benchmark suite..", fg='green')
+    click.secho(f"\n\nConfigured execution parameters are:", fg='green')
+    click.secho(f"{execution_params}", fg="blue")
+
+    # Setup execution env
+    if not ts:
+        if execution_params['exec_env'] is 'local':
+            click.secho("\n\nSetting up local execution..", fg='green')
+            local_torserve_start()
+        else:
+            click.secho("\n\nSetting up docker execution..", fg='green')
+            docker_torchserve_start()
+    exec_ab_benchmark()
+    report_generate()
+
+
+def exec_ab_benchmark():
+    click.secho("*Registering model...", fg='green')
+    RURL = f"?model_name=benchmark&url={execution_params['url']}&batch_size={execution_params['batch_size']}&" \
+           f"max_batch_delay={execution_params['batch_delay']}&initial_workers={execution_params['workers']}&synchronous=true"
+    execute(f"curl -X POST \"http://localhost:8081/models{RURL}\"", wait=True)
+
+    click.secho("\n\nExecuting Apache Bench tests ...", fg='green')
+    click.secho("*Executing inference performance test", fg='green')
+    ab_cmd = f"ab -c {execution_params['concurrency']}  -n {execution_params['requests']} -k -p /tmp/benchmark/input -T " \
+             f"{execution_params['content_type']} http://127.0.0.1:8080/predictions/benchmark > {result_file}"
+    execute(ab_cmd, wait=True)
+
+    click.secho("*Unregistering model ...", fg='green')
+    execute("curl -X DELETE \"http://localhost:8081/models/benchmark\"", wait=True)
+    click.secho("*Terminating Torchserve instance...", fg='green')
+    execute("torchserve --stop", wait=True)
+    click.secho("Apache Bench Execution completed.", fg='green')
```

Comment:
Use python requests module instead of curl

Reply:
Done